src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesVsPatterns.cs

```diff
@@ -1927,12 +1924,12 @@ public interface IOut<out T> { }
                 // (40,29): warning CS8619: Nullability of reference types in value of type 'I<string?>' doesn't match target type 'I<string>'.
                 //         _ = i switch { 1 => y, _ => x }/*T:I<string!>!*/; // 4
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "y").WithArguments("I<string?>", "I<string>").WithLocation(40, 29),
-                // (45,37): warning CS8653: A default expression introduces a null value when 'T' is a non-nullable reference type.
+                // (45,37): warning CS8601: Possible null reference assignment.
                 //         _ = i switch { 1 => x, _ => default }/*T:T*/; // 5
-                Diagnostic(ErrorCode.WRN_DefaultExpressionMayIntroduceNullT, "default").WithArguments("T").WithLocation(45, 37),
-                // (46,29): warning CS8653: A default expression introduces a null value when 'T' is a non-nullable reference type.
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(45, 37),
```

Comment:
>WRN_NullReferenceAssignment [](start = 37, length = 27)

Seems questionable.  Shouldn't discards always be capable of receiving null?

Reply:
Logged [https://github.com/dotnet/roslyn/issues/39888](https://github.com/dotnet/roslyn/issues/39888)

---
In reply to: [347696516](https://github.com/dotnet/roslyn/pull/39778#discussion_r347696516) [](ancestors = 347696516)